test/test-parse-expression.cpp

```diff
@@ -3161,6 +3161,273 @@ TEST_F(test_parse_expression,
   }
 }
 
+TEST(test_parse, test_expression_for_potential_side_effects) {
+  {
+    test_parser p(u8"class {};"_sv);
+    expression* ast = p.parse_expression();
+    bool side_effects = p.parser().has_potential_side_effects(ast);
+    EXPECT_TRUE(side_effects);
+  }
```

Comment:
This really should be `false`, but I know we can't implement that right now. Perhaps the following examples are better, because they do have side effects:
```js
class { [f()]() { } }
```
or:
```js
class { static pi = calculatePi(); }
```

Reply:
This one is resolved.